lib/web/terminal.go

```diff
@@ -112,21 +112,27 @@ func NewTerminal(req TerminalRequest, authProvider AuthProvider, ctx *SessionCon
 		return nil, trace.BadParameter("invalid server name %q: %v", req.Server, err)
 	}
 
-	return &TerminalHandler{
+	terminalHanlder := newTerminalHandler()
```

Comment:
Since you have everything you need to create the terminal handler, why create it then set the fields individually? Why not create the struct yourself or have a `New` function that takes a `Config`. Examples below.

```golang
handler := NewTerminalHandler{
   Foo: "",
   Bar: ""
}
```

```golang
handler := NewTerminalHandler(&Config{
   Foo: "",
   Bar: "",
})
```

Reply:
> why create it then set the fields individually? 

I added a little factory method to isolate encoder/decoder initialization from the rest of the parameters. This is to make it easer for the unit-tests although it would be better to create that helper in unit-tests to avoid this confusing. 

I thought of adding a configuration object but decided to keep the scope of the changes small.

Reply:
This does not look consistent with the style we use in the codebase, which is outlined by @russjones. Just use the options structure and set defaults for encoder and decoder.